.github/workflows/build-wheels-upload-to-pypi.yml

```diff
@@ -0,0 +1,37 @@
+name: Build and Push to PyPI
+
+on:
+  release:
+    types: [published]
+
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, windows-2019, macOS-10.15]
+        python-version: [3.6, 3.7, 3.8, 3.9]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      # Used to host cibuildwheel
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install cibuildwheel
+        run: python -m pip install cibuildwheel
+
+      - name: Build wheels
+        run: python -m cibuildwheel --output-dir wheelhouse
```

Comment:
I think we would also like to run all tests on the built wheel before uploading it to PyPI.

Reply:
Great, `cibuildhweel` exposes configuration options to run tests on the wheels, I tested a few options but couldn't make it work yet. Probably need a bit more massaging.